compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1170,7 +1170,7 @@ class Typer extends Namer
       if (tree.tpt.isEmpty)
         meth1.tpe.widen match {
           case mt: MethodType =>
-            pt match {
+            pt.stripAnnots match {
```

Comment:
I think the stripping should be done in the `SAMType` extractor instead

Reply:
Indeed. Moved.